include/cuco/detail/extent/extent.inl

```diff
@@ -30,11 +30,12 @@ template <typename SizeType, std::size_t N>
 struct window_extent {
   using value_type = SizeType;  ///< Extent value type
 
+  __host__ __device__ explicit constexpr window_extent() noexcept {}
```

Comment:
Dynamic `window_extent` is a template specialization which has has the ctor still private. I just enabled the default ctor for the static case. This make the static shmem usecase a bit easier, since you can get the actual extent by calling `typename ref_type::extent_type{}`.

Reply:
Oh, wow, I missed that. Although the dynamic extent isn't the problem here since it's a different specialization with the ctor still being private. The problem is that a user can specify any size and it will be assumed as a valid size. I'll roll back this commit.